### PR TITLE
port tests for F.pad and transforms.Pad

### PR DESCRIPTION
**Pull request by pmeier**
cc @vfdev-5

### CODE REVIEW

File: test/test_transforms_v2_refactored.py [hunk 1]
```diff
@@ -3346,3 +3344,169 @@ def test_transform_errors_warnings(self):
...
+            F.pad_image,
+            image,
+            **kwargs,
+            check_scripted_vs_eager=not (
+                (param == "padding" and isinstance(value, int))
+                # See https://github.com/pytorch/vision/pull/7252#issue-1585585521 for details
+                or (
+                    param == "fill"
+                    and (
+                        isinstance(value, tuple) or (isinstance(value, list) and all(isinstance(v, int) for v in value))
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:3384]: Curious if, strictly speaking, `all` should be `any`?

(it doesn't make a difference in those tests given the current values of `EXHAUSTIVE_TYPE_FILLS`)

File: test/test_transforms_v2_refactored.py [hunk 2]
```diff
@@ -3346,3 +3344,169 @@ def test_transform_errors_warnings(self):
...
+        [make_image_tensor, make_image_pil, make_image, make_bounding_boxes, make_segmentation_mask, make_video],
+    )
+    def test_functional(self, make_input):
+        check_functional(F.pad, make_input(), padding=[1])
+
+    @pytest.mark.parametrize(
+        ("kernel", "input_type"),
+        [
+            (F.pad_image, torch.Tensor),
+            # FIXME: The PIL kernel uses fill=0 as default rather than fill=None as all others
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:3433]: Is this ever going to be fixed / fixable?

**Review by NicolasHug — APPROVED** [review]
Thanks Philip

File: test/test_transforms_v2_refactored.py (hunk 2 above)
Comment (pmeier) [reply test/test_transforms_v2_refactored.py:3433]: We could just put

https://github.com/pytorch/vision/blob/e9f80941358f2d15f7138667caaf07beb2f9b3fc/torchvision/transforms/v2/functional/_geometry.py#L1170-L1171

into the PIL kernel as well.

**Pull request merged**